docs/user-guide.md

```diff
@@ -362,6 +362,42 @@ spec:
 Note that the mutating admission webhook is needed to use this feature. Please refer to the 
 [Quick Start Guide](quick-start-guide.md) on how to enable the mutating admission webhook.
 
+### Python Support
+
+Python support can be enabled by setting `.spec.mainApplicationFile` with path to your python application.
+Optionaly, `.spec.pythonVersion` parameter can be used to set the major Python version of the docker image used 
+to run the driver and executor containers. Below is an example showing part of a `SparkApplication` specification:
+
+```yaml
+spec:
+  type: Python
+  pythonVersion: 2
+  mainApplicationFile: local:///opt/spark/examples/src/main/python/pyfiles.py
+```
+
+Some PySpark applications need additional Python packages to run. 
+Such dependencies are specified using the optional field `.spec.deps.pyFiles` , which translates to the --py-files 
+option of the spark-submit command.
+
+```yaml
+spec:
+  deps:
+    pyfiles:
+       - local:///opt/spark/examples/src/main/python/py_container_checks.py
+       - gs://spark-data/python-dep.zip
+``` 
+
+In order to use the dependencies that are hosted remotely, the following PySpark code
+can be used in Spark 2.4.
+  
+```
+python_dep_file_path = SparkFiles.get("python-dep.zip")
+spark.sparkContext.addPyFile(dep_file_path)
+``` 
+
+Note that Python binding for PySpark will available in Apache Spark 2.4, 
+and currently requires building custom 2.4.0-SNAPSHOT Docker image.
```

Comment:
`currently requires building a custom Docker image from the Spark master branch.`.